### PR TITLE
[lint] replaces black formatting with ruff

### PR DESCRIPTION
**Pull request by iris-garden**
`ruff` is faster than `black` when formatting Python `files`, so this change replaces the formatting checks for our files with calls to `ruff` instead of `black`, and runs `ruff format` on all files that would cause those checks to fail.

### CODE REVIEW

**Comment by iris-garden** [comment]
oh wait, okay, i'm gonna see if there's an equivalent to `skip-string-normalization` for `ruff`, since i think that might be literally all of these automatic changes haha

**Comment by iris-garden** [comment]
okay, enabled the `quote-style = "preserve"` flag, now this should be actually ready for a look!

**Comment by daniel-goldstein** [comment]
Interesting, [PEP 8](https://peps.python.org/pep-0008/#string-quotes) does have an opinion about triple-quoted strings. FWIW, I'm happy to go stricter in the future and allow string normalization and all that, but I appreciate you aligning the configs for this PR!

File: hail/python/hail/experimental/datasets.py [hunk 1]
```diff
@@ -77,41 +79,42 @@ def load_dataset(name: str,
...
+            f' {reference_genomes}.'
+        )
+
+    clouds = set(k for dataset in datasets[name]['versions'] for k in dataset['url'].keys())
     if cloud not in clouds:
-        raise ValueError(f'Cloud platform {repr(cloud)} not available for'
-                         f' dataset {name}.\n'
-                         f'Available platforms: {clouds}.')
+        raise ValueError(
+            f'Cloud platform {repr(cloud)} not available for' f' dataset {name}.\n' f'Available platforms: {clouds}.'
```
Comment (daniel-goldstein) [inline hail/python/hail/experimental/datasets.py:102]: This line is a little unfortunate. Let's combine them into one f-string.

**Review by daniel-goldstein — DISMISSED** [review]
This looks great! One last request and I'll approve: could you remove `black` from the dev requirements and replace the pre-commit hook?

File: hail/python/hail/experimental/datasets.py (hunk 1 above)
Comment (iris-garden) [reply hail/python/hail/experimental/datasets.py]: done!

**Comment by iris-garden** [comment]
updated the pre-commit hook, and in the process realized i had only run `ruff format` on files that are checked by `make check-all`, so also ran it at the root of the repo, as the hook will run there

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**